sbnci/Modules/PDS/PDSValidation_module.cc

```diff
@@ -0,0 +1,346 @@
+  fTree->Branch("ArapucaArea",&fArapucaArea);
+  fTree->Branch("ArapucaAmplitude",&fArapucaAmplitude);
+  fTree->Branch("ArapucaPE",&fArapucaPE);
+  fTree->Branch("ArapucaFastToTotal",&fArapucaFastToTotal);
+
+
+//  fTimeHist=tfs->make<TH1F>("fTimeHist", ";FlashTime (ns)", 20, 0, 100);
+}// end beginJob
+
+////////////////////////////////////////////////////////////////////
+void ana::PDSValidation::analyze(const art::Event& evt) {
+
+  fRun    = evt.run();
+  fSubRun = evt.subRun();
+  fEvent  = evt.event();
+
+  if(fVerbose) {
+      cout << "Analysing run " << fRun << ", subrun " << fSubRun << ", event: " << fEvent << endl;
+  }
+  numevents++;
+
+  //Getting  MC truth information
+  art::Handle< vector<simb::MCTruth> > mctruthListHandle;
+  vector<art::Ptr<simb::MCTruth> > mclist;
+  if(evt.getByLabel(fGenieGenModuleLabel,mctruthListHandle)){
+      art::fill_ptr_vector(mclist, mctruthListHandle);
+  }
+
+  //###############################################
+  //### Get the Truth information for the event ###
+  //###############################################
+
+  //List the particles in the event
+  const sim::ParticleList& particles = particleInventory->ParticleList();
+
+  //Loop over the particles
+  map<int,const simb::MCParticle*> trueParticles;
+  map<int,bool> mcparticlescontained;
+  map<int,float> trueParticleEnergy;
+
+  //auto const clockData = art::ServiceHandle<detinfo::DetectorClocksService const>()->DataFor(evt);
+
+  //Make a map of Track id and pdgcode
+  for (const auto& particleIt: particles) {
+
+      const simb::MCParticle* particle = particleIt.second;
+      trueParticles[particle->TrackId()] = particle;
+
+      if(fVerbose){
+          cout << "True Particle with track ID: " << particle->TrackId() << " Has code of: " 
+               << particle->PdgCode() << " and Energy of: " << particle->E() << " With Mother: " 
+               << particle->Mother() << " Proccess: " << particle->Process() << " End Process: "  
+               << particle->EndProcess() 
+          << endl;
+      }// if verbose
+
+  } // for MCParticles
+
+
+  //###############################################
+  //### Get PDS reco info for the event         ###
+  //###############################################
+  //Getting  OpFlash information
+  art::Handle< vector<recob::OpFlash> > opFlashListHandle;
+  vector<art::Ptr<recob::OpFlash> > opFlashList;
+  if(evt.getByLabel(fOpFlashTPC1Label, opFlashListHandle)){
+      art::fill_ptr_vector(opFlashList, opFlashListHandle);
+  }
+
+  fNFlash = opFlashList.size();
+
+  for(auto const& flash : opFlashList) {
+    fFlashTotalPE.push_back(flash->TotalPE());
+    fFlashTime.push_back(flash->Time());
+    fFlashTimeWidth.push_back(flash->TimeWidth());
+    fYCenter.push_back(flash->YCenter());
+    fZCenter.push_back(flash->ZCenter());
+    fYWidth.push_back(flash->YWidth());
+    fZWidth.push_back(flash->ZWidth());
+//    fTimeHist->Fill(flash->Time());
```

Comment:
As above

Reply:
probably keep for now. it's something for the experts to look at.